sdmetrics/reports/multi_table/diagnostic_report.py

```diff
@@ -76,12 +79,15 @@ def generate(self, real_data, synthetic_data, metadata):
                 The metadata, which contains each column's data type as well as relationships.
         """
         metrics = list(itertools.chain.from_iterable(self.METRICS.values()))
+        self._metric_args['NewRowSynthesis']['synthetic_sample_size'] = min(
+            len(real_data), self._metric_args['NewRowSynthesis']['synthetic_sample_size'])
 
         for metric in tqdm.tqdm(metrics, desc='Creating report'):
             metric_name = metric.__name__
             try:
+                args = self._metric_args.get(metric_name, {})
```

Comment:
The parameter `args` confuses me since it's a special key or in some cases it can be reserved for the method definition eg: `def func(*args, **kwargs)` .I would suggest to change `args` to something like `metric_args` or similar just to avoid this type of confusion.